### PR TITLE
fix: emit by order

### PR DESCRIPTION
**Pull request by o-rumiantsev**
Closes: https://github.com/o-rumiantsev/Dirsync/issues/11

### CODE REVIEW

**Comment by o-rumiantsev** [comment]
Blocked by https://github.com/o-rumiantsev/Dirsync/pull/10

File: lib/connection.js [hunk 1]
```diff
@@ -55,6 +65,33 @@ class Connection extends EventEmitter {
...
+    }
+    this.emit('packet', packet);
+    ++this[kNextId];
+    this[walkPool]();
+  }
+
+  [walkPool]() {
+    const packet = this[kPool].get(this[kNextId]);
+    if (packet) {
+      this.emit('packet', packet);
```
Comment (mille-nium) [inline lib/connection.js]: Kind of boilerplate code, same as in `processPacket`, so you can simply write here 
```javascript
this.processPacket(packet)
```

File: lib/connection.js [hunk 2]
```diff
@@ -55,6 +65,33 @@ class Connection extends EventEmitter {
     }
   }
 
+  processPacket(packet) {
```
Comment (mille-nium) [inline lib/connection.js]: This should be a private method

File: lib/connection.js [hunk 3]
```diff
@@ -55,6 +65,33 @@ class Connection extends EventEmitter {
...
+    if (packet.id !== this[kNextId]) {
+      this[kPool].set(packet.id, packet);
+      return;
+    }
+    this.emit('packet', packet);
```
Comment (mille-nium) [inline lib/connection.js]: Should emit only payload, FWIW

**Comment by mille-nium** [comment]
Also add comments, please

File: lib/connection.js (hunk 3 above)
Comment (mille-nium) [reply lib/connection.js]: And probably rename event to `message` or smth like this

File: lib/connection.js [hunk 4]
```diff
@@ -46,18 +58,43 @@ class Connection extends EventEmitter {
...
+    }
+    const data = packet.payload.toString() || '{}';
+    const msg = JSON.parse(data);
+    this.emit('message', msg);
+    ++this[kNextId];
+    this[walkPool]();
+  }
+
+  // Walk through the pool of packets to find next expected packet
+  [walkPool]() {
```
Comment (mille-nium) [inline lib/connection.js]: Maybe, `inspectPool` would be more obvious. After changing can be landed

**Review by mille-nium — CHANGES_REQUESTED** [review]

**Review by mille-nium — APPROVED** [review]

**Pull request merged**